tracing/src/main/java/com/palantir/tracing/Span.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.tracing;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Optional;
+import org.immutables.value.Value;
+
+/**
+ * A value class representing a completed Span.
+ */
+@JsonDeserialize(as = ImmutableSpan.class)
+@JsonSerialize(as = ImmutableSpan.class)
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
+public abstract class Span {
+
+    public abstract String getTraceId();
+
+    public abstract Optional<String> getParentSpanId();
+
+    public abstract String getSpanId();
+
+    public abstract String getOperation();
+
+    public abstract long getStartTimeMs();
+
+    public abstract long getDurationNs();
```

Comment:
Accurate timing on the JVM needs to use the nanosecond counter on the CPU, but wall clock time is only really available in milliseconds. I'm happy to convert the nanoseconds to milliseconds but I figured I'd save the math operation and leave that to a subscriber. (Don't have a strong opinion, just explaining rationale.)